crypto/property/property_string.c

```diff
@@ -244,3 +245,33 @@ const char *ossl_property_value_str(OSSL_LIB_CTX *ctx, OSSL_PROPERTY_IDX idx)
 {
     return ossl_property_str(0, ctx, idx);
 }
+
+/*
+ * Take an existing propquery string, and add a property to restrict the fetch
+ * to a given provider
+ */
+char *ossl_property_str_for_provider(const char *propquery,
+                                     const OSSL_PROVIDER *prov)
```

Comment:
This is a curious location for this function...  it would be more suitable in provider_core.c or in its own file...

Reply:
I was struggling to figure out where the best location was. Happy to move it to provider_core.c

Reply:
It definitely shouldn't be here.  `property_string.c` is a very low level cache mapping strings to small integers.